tensornetwork/backends/tensorflow/tensorflow_backend.py

```diff
@@ -83,7 +83,10 @@ def diag(self, tensor: Tensor) -> Tensor:
     return self.tf.linalg.diag(tensor)
 
   def convert_to_tensor(self, tensor: Tensor) -> Tensor:
-    result = self.tf.convert_to_tensor(tensor)
+    if isinstance(tensor, (float, int)):
```

Comment:
what is this case covering?

Reply:
Given a scalar, we need to specify the data type to create a tensor in tensorflow.

Reply:
Just worried that explicit use of `tf.float64` could lead to problems if other dtypes are e.g. tf.float32

Reply:
hmm, the input is constrained to float and int in python type, could that protect tf.float64?

Reply:
Can you tell me why you added this?

Reply:
self.tf.convert_to_tensor(tensor) throws an error if the input is float/int.

Reply:
OK, but where in the code do you need to pass a float or int to backed.convert_to_tensor?

Reply:
Following functions call backed.convert_to_tensor.  Test case is like np.ones((1, 2, 3)) + 2

def __sub__()
def __add__()